nano/node/active_transactions.cpp

```diff
@@ -67,99 +67,152 @@ void nano::active_transactions::stop ()
+{
+	auto status = election->status;
+	auto votes = election->votes_with_weight ();
+
+	node.observers.blocks.notify (status, votes, account, amount, is_state_send, is_state_epoch);
+
+	if (amount > 0)
+	{
+		node.observers.account_balance.notify (account, false);
+		if (!pending_account.is_zero ())
 		{
-			node.ledger.cache.final_votes_confirmation_canary.store (true);
+			node.observers.account_balance.notify (pending_account, true);
 		}
+	}
+}
 
-		// Next-block activations are done after cementing hardcoded bootstrap count to allow confirming very large chains without interference
-		bool const cemented_bootstrap_count_reached{ node.ledger.cache.cemented_count >= node.ledger.bootstrap_weight_max_blocks };
+void nano::active_transactions::handle_final_votes_confirmation (std::shared_ptr<nano::block> const & block, nano::store::read_transaction const & transaction, nano::election_status_type status)
+{
+	auto const & account = !block->account ().is_zero () ? block->account () : block->sideband ().account;
 
-		// Next-block activations are only done for blocks with previously active elections
-		bool const was_active{ *election_status_type == nano::election_status_type::active_confirmed_quorum || *election_status_type == nano::election_status_type::active_confirmation_height };
+	bool is_canary_not_set = !node.ledger.cache.final_votes_confirmation_canary.load ();
+	bool is_canary_account = account == node.network_params.ledger.final_votes_canary_account;
+	bool is_height_above_threshold = block->sideband ().height >= node.network_params.ledger.final_votes_canary_height;
 
-		if (cemented_bootstrap_count_reached && was_active)
-		{
-			// Start or vote for the next unconfirmed block
-			node.scheduler.priority.activate (account, transaction);
+	if (is_canary_not_set && is_canary_account && is_height_above_threshold)
+	{
+		node.ledger.cache.final_votes_confirmation_canary.store (true);
+	}
 
-			// Start or vote for the next unconfirmed block in the destination account
-			auto const & destination (node.ledger.block_destination (transaction, *block_a));
-			if (!destination.is_zero () && destination != account)
-			{
-				node.scheduler.priority.activate (destination, transaction);
-			}
-		}
+	bool cemented_bootstrap_count_reached = node.ledger.cache.cemented_count >= node.ledger.bootstrap_weight_max_blocks;
+	bool was_active = status == nano::election_status_type::active_confirmed_quorum || status == nano::election_status_type::active_confirmation_height;
+
+	if (cemented_bootstrap_count_reached && was_active)
+	{
+		activate_scheduler_for_account_and_destination (account, block, transaction);
+	}
+}
+
+void nano::active_transactions::activate_scheduler_for_account_and_destination (const nano::account & account, std::shared_ptr<nano::block> const & block, nano::store::read_transaction const & transaction)
```

Comment:
This name is a bit long. Maybe "activate_dependents".

Reply:
I agree that it's too long. Since it activates the next blocks, what about "activate_successors" ?